cli/src/migrations/files.ts

```diff
@@ -29,7 +30,11 @@ async function readMigrationsDir() {
   }
 }
 
-export async function getLocalMigrationFiles(): Promise<Schemas.MigrationObject[]> {
+export type LocalMigrationFile =
+  | Schemas.MigrationObject
+  | (Schemas.PgRollMigrationHistoryItem & { id?: never; checksum?: never; operations?: never[] });
```

Comment:
What does the `never` value do in this case? Isn't this the default for `Schemas.PgRollMigrationHistoryItem`?

Reply:
So, in a union `A | B` you either have the properties of `A` or the properties of `B`. Trying to access `B.id` fails because `id` is not a property of `B`, however if we add an intersection of `never` (something that is never going to happen), `B.id` is now a property of `B` although you can never read or assign a value. In your code, it allowed to remove all the `as` casting trying to read `(A | B).id`.

Reply:
I see, thank you!